lib/src/formats/jpeg/jpeg_data.dart

```diff
@@ -141,7 +141,8 @@ class JpegData {
     if (frame != null) {
       info
         ..width = frame!.samplesPerLine!
-        ..height = frame!.scanLines!;
+        ..height = frame!.scanLines!
+        ..numComponents = frame!.components.length;
```

Comment:
Technically these frame.components are the internal Y',Cb,Cr encoded color information from the Jpeg, not the decoded RGB components of the resulting image.

In this case, it happens that they are both 3. But a JPEG will always decode to an RGB 3-channel image, so you could just set numComponents to 3.

Reply:
As I am using this information to check if we can directly insert the jpeg into a PDF (which natively supports jpeg), I skip the decoding step. I do need the information to setup the image in pdf though, so the number of channels is required outside of the actual jpeg image.